frontend/src/components/App/Settings/DrawerModeButton.tsx

```diff
@@ -0,0 +1,55 @@
+import { FormControlLabel, Switch } from '@mui/material';
+import React from 'react';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { setDetailDrawerEnabled } from '../../../redux/drawerModeSlice';
+import { useTypedSelector } from '../../../redux/reducers/reducers';
+import { TooltipIcon } from '../../common';
+
+export default function DrawerModeButton() {
```

Comment:
Let's name this DrawerModeSettings. There's more here than a button

Reply:
done!